grammar.js

```diff
@@ -1072,7 +1072,7 @@ module.exports = grammar({
     instance_variable: $ => token(seq('@', ALPHA_CHAR, IDENTIFIER_CHARS)),
     class_variable: $ => token(seq('@@', ALPHA_CHAR, IDENTIFIER_CHARS)),
 
-    global_variable: $ => /\$-?(([!@&`'+~=/\\,;.<>*$?:"])|([0-9]*)|([a-zA-Z_][a-zA-Z0-9_]*))/,
+    global_variable: $ => /\$(-[a-zA-Z0-9_]|[!@&`'+~=/\\,;.<>*$?:"]|[0-9]+|[a-zA-Z_][a-zA-Z0-9_]*)/,
```

Comment:
Would it be possible to add a test for this to see what new cases are supported by this change?

Reply:
It's not really new cases, after this change we actually recognise fewer cases. Some examples are included in:
https://github.com/tree-sitter/tree-sitter-ruby/pull/225/commits/b38ceb586020f24e5a4c5379d97a6dc75c1a3d49#diff-c8a6ceb55807f1b0bde1cb7f3951f8708c04941e25d7aa7eee3c943de1fb12acR588-R597

Reply:
Thanks for clarifying that!